--- include/extern.h	2017-12-12 19:42:52.025862258 -0500
+++ include/extern.h	2017-12-13 13:47:04.214207665 -0500
@@ -2769,6 +2769,7 @@
 E int FDECL(destroy_mitem, (struct monst *, int, int));
 E int FDECL(resist, (struct monst *, CHAR_P, int, int));
 E void NDECL(makewish);
+E void NDECL(blessitem);
 
 #endif /* !MAKEDEFS_C && !LEV_LEX_C */
 
--- src/engrave.c	2017-12-12 19:42:52.029195636 -0500
+++ src/engrave.c	2017-12-13 14:13:20.364925487 -0500
@@ -645,6 +645,7 @@
             case WAN_CREATE_MONSTER:
             case WAN_WISHING:
             case WAN_ENLIGHTENMENT:
+            case WAN_BLESSING:
                 zapnodir(otmp);
                 break;
             /* IMMEDIATE wands */
--- src/invent.c	2017-12-12 19:42:52.029195636 -0500
+++ src/invent.c	2017-12-13 14:13:39.755335978 -0500
@@ -1100,6 +1100,7 @@
                      || !objects[HORN_OF_PLENTY].oc_name_known))
              || (!strcmp(word, "charge") && !is_chargeable(otmp))
              || (!strcmp(word, "call") && !objtyp_is_callable(otyp))
+             || (!strcmp(word, "bless") && (otmp->blessed || !otmp->bknown))
              ) {
                 foo--;
             }
--- src/mkobj.c	2017-12-12 19:42:52.032529014 -0500
+++ src/mkobj.c	2017-12-13 14:07:52.000000000 -0500
@@ -971,7 +971,7 @@
             }
             break;
         case WAND_CLASS:
-            if (otmp->otyp == WAN_WISHING)
+            if (otmp->otyp == WAN_WISHING || otmp->otyp == WAN_BLESSING)
                 otmp->spe = rnd(3);
             else
                 otmp->spe =
--- src/objects.c	2017-12-12 19:42:52.032529014 -0500
+++ src/objects.c	2017-12-13 13:43:51.898189185 -0500
@@ -1022,7 +1022,8 @@
 WAND("cold",            "short", 40, 175, 1, RAY, IRON, HI_METAL),
 WAND("sleep",           "runed", 50, 175, 1, RAY, IRON, HI_METAL),
 WAND("death",            "long",  5, 500, 1, RAY, IRON, HI_METAL),
-WAND("lightning",      "curved", 40, 175, 1, RAY, IRON, HI_METAL),
+WAND("lightning",      "curved", 35, 175, 1, RAY, IRON, HI_METAL),
+WAND("blessing",    "sparkling",  5, 500, 1, NODIR, IRON, HI_METAL),
 /* extra descriptions, shuffled into use at start of new game */
 WAND(None,             "forked",  0, 150, 1, 0, WOOD, HI_WOOD),
 WAND(None,             "spiked",  0, 150, 1, 0, IRON, HI_METAL),
--- src/zap.c	2017-12-12 19:42:52.032529014 -0500
+++ src/zap.c	2017-12-13 14:14:04.982528408 -0500
@@ -2039,6 +2039,10 @@
         pline_The("feeling subsides.");
         exercise(A_WIS, TRUE);
         break;
+    case WAN_BLESSING:
+        known = TRUE;
+        blessitem();
+        break;
     }
     if (known) {
         if (!objects[obj->otyp].oc_name_known)
@@ -4921,6 +4925,32 @@
 }
 
 void
+blessitem()
+{
+    char allowall[2];
+    struct obj *otmp;
+
+    allowall[0] = ALL_CLASSES;
+    allowall[1] = '\0';
+
+    if(flags.verbose)
+        You("may bless an object in your inventory.");
+
+    otmp = getobj(allowall, "bless");
+
+    if(!otmp) {
+        return;
+    }
+
+    else if (otmp != &zeroobj) {
+        otmp->blessed = 1;
+        otmp->bknown = 1;
+        pline("%s%s%s for a moment.", Yobjnam2(otmp, Blind ? "vibrate" : "glow"),
+            Blind ? "" : " ", Blind ? "" : hcolor(NH_BLUE));
+    }
+}
+
+void
 makewish()
 {
     char buf[BUFSZ], promptbuf[BUFSZ];
